apps/gazettes-aleph/gazettes-aleph.yml

```diff
@@ -0,0 +1,111 @@
+---
+- hosts:
+    - gazettes-ansible-test-openup.org.za
+  become: yes
+  roles:
+   - dokku_bot.ansible_dokku
+
+  vars:
+    app_name: "gazettes-aleph-{{ env_name }}"
+    app_domain: "{{ inventory_hostname }}"
+    dokku_daemon_install: false # We install this using a dedicated playbook
+    dokku_plugins:
+      - name: letsencrypt
+        url: https://github.com/dokku/dokku-letsencrypt.git
+      - name: elasticsearch
+        url: https://github.com/dokku/dokku-elasticsearch.git
+    postgres_hostname: postgresql11-prod.cnc362bhpvfe.eu-west-1.rds.amazonaws.com
+    postgres_superuser_user: "{{ lookup('passwordstore', 'core/aleph/POSTGRES_SUPERUSER subkey=username')}}"
+    postgres_superuser_password: "{{ lookup('passwordstore', 'core/aleph/POSTGRES_SUPERUSER')}}"
+    postgres_user: "{{ lookup('passwordstore', 'apps/aleph/gazettes-aleph/{{ env_name }}/POSTGRES subkey=username')}}"
+    postgres_password: "{{ lookup('passwordstore', 'apps/aleph/gazettes-aleph/{{ env_name }}/POSTGRES')}}"
+
+  tasks:
+    - name: "Dokku app exists"
+      dokku_app:
+        app: "{{ app_name }}"
+
+    - name: "Dokku elasticsearch"
+      vars:
+        ELASTICSEARCH_IMAGE: "openup/elasticsearch-0.90"
+        ELASTICSEARCH_IMAGE_VERSION: "latest"
+      shell: "dokku elasticsearch:create openup-es"  # Is there a way to do this using dokku plugin?
+
+    - name: "Dokku app config is set"
+      dokku_config:
+        app: "{{ app_name }}"
+        config:
+          ALEPH_APP_NAME: "{{ app_name }}"
+          ALEPH_APP_TITLE: "Open Gazettes South Africa"
+          ALEPH_APP_URL: "https://search.opengazettes.org.za"
+          ALEPH_ARCHIVE_BUCKET: "code4sa-aleph"
+          ALEPH_ARCHIVE_TYPE: "s3"
+          ALEPH_BROKER_URI: "sqs://sqs.eu-west-1.amazonaws.com/567304594100/"
+          ALEPH_DATABASE_URI: "postgresql://{{ postgres_user }}:{{ postgres_password }}@{{ postgres_hostname }}:5432/{{ postgres_database }}"
+          # This should be an unexposed docker container on an internal docker network, make sure all systems that need access are attached to network
+          ALEPH_ELASTICSEARCH_URI: "http://openup-es:9200/"
```

Comment:
don't set this here - either configure dokku:elasticsearch to set this variable (if it can set a custom variable name) or set it manually from the DSN you get when creating the instance

Reply:
@jbothma I don't understand why you wouldn't want to set this here

Reply:
because you don't know what DSN dokku-elasticsearch is going to generate for you

```
jdb@gazettes-aleph-test:~$ dokku elasticsearch:info openup-es
=====> openup-es elasticsearch service information
       Config dir:          /var/lib/dokku/services/elasticsearch/openup-es/config
       Data dir:            /var/lib/dokku/services/elasticsearch/openup-es/data
       Dsn:                 http://dokku-elasticsearch-openup-es:9200
       Exposed ports:       -                        
       Id:                  6a0166266f73216bb56dca3d55347fba7ac293c9da8e3ab1ccc4697bdac54d0c
       Internal ip:         172.17.0.2               
       Links:               -                        
       Service root:        /var/lib/dokku/services/elasticsearch/openup-es
       Status:              running                  
       Version:             elasticsearch:6.8.5      
```

Reply:
what you could do is link them
```
dokku elasticsearch:link openup-es gazettes-aleph
```

and then set the result of `dokku elasticsearch:info openup-es --dsn` as this variable

Reply:
it looks like it allows the link to be created (well, the config for the link because the app container is ephemeral) before pushing the app code

Reply:
it sets `ELASTICSEARCH_URL` by default on the app and the plugin doesn't allow the variable name to be customised yet.

Reply:
@jbothma 

This plugin seems to overcomplicate things, especially if one day we would move to ES hosted externally.

If dokku's containers are part of the default network (or if not, perhaps we can define one that dokku attaches to its containers), then can just get ansible to ensure elasticsearch is present, attached to that network .